src/winforms/toga_winforms/widgets/base.py

```diff
@@ -8,6 +8,9 @@ def __init__(self, interface):
 
         self._container = None
         self.native = None
+        # `winforms_event_handlers` is used to keep track of added
+        # native event handlers to remove them when a widget is removed
+        self.winforms_event_handlers = []
```

Comment:
I agree this is a good change; I wonder if it might be worth going "all in" here, and writing a wrapper API for managing these events? e.g., instead of expecting the user to use the Python.NET API to add the event, and *then* make sure it's added to the handler list, expose a single (toga-winforms-specific) `add_winforms_handler()` method that both registers the handler *and* adds it to the list to be managed.

Reply:
I was sure that you would suggest a more elegant solution :) 
I will work on adding the `add_winforms_handler` after figuring out the garbage collection via test app.

Reply:
:-) 

To be clear, though - `add_winforms_handler()` may end up not being needed at all - if Python.NET *does* do full garbage collection, then the sort of event handling we have at present (i.e., before this patch - just assigning the event hander) will be more than sufficient, as there's no need to do a coordinated clean up on deletion.

Reply:
I've removed all `add_winforms_handler` calls as they seem unnecessary at the moment.